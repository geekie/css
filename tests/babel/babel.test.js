"use strict";

const babel = require("@babel/core");
const fs = require("fs");
const path = require("path");
const atomic = require("../../atomic");
const plugin = require("../../babel");

function compile(input) {
  const { code, metadata } = babel.transform(input, { plugins: [plugin] });
  return { code, metadata };
}

beforeEach(atomic.reset);

const separator = "\n      ↓ ↓ ↓ ↓ ↓ ↓\n\n";

const files = fs
  .readdirSync(path.resolve(__dirname, "fixtures"))
  .filter(f => /\.js$/.test(f));

describe.each(files)("%s", filename => {
  const realpath = path.resolve(__dirname, path.join("fixtures", filename));

  test.each(["prod", "dev"])("(%s)", env => {
    const source = fs.readFileSync(realpath);
    const { code: output, metadata } = babel.transform(source, {
      filename: realpath,
      plugins: [[plugin, { prettify: env === "dev" }]]
    });
    expect(source + separator + output).toMatchSnapshot("compiled");
    expect(metadata.gkcss.css).toMatchSnapshot("css");
    expect(metadata.gkcss.dependencies).toMatchSnapshot("dependencies");
  });
});
