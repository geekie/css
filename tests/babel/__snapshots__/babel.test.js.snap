// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function.js (dev): compiled 1`] = `
function getGreen() {
  return "green";
}

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: getGreen()
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

function getGreen() {
  return "green";
}

const styles = {
  bgGreen: "gkcss0-backgroundColor-green"
};
`;

exports[`function.js (dev): css 1`] = `.gkcss0-backgroundColor-green{background-color:green}`;

exports[`function.js (dev): dependencies 1`] = `Array []`;

exports[`function.js (prod): compiled 1`] = `
function getGreen() {
  return "green";
}

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: getGreen()
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

function getGreen() {
  return "green";
}

const styles = {
  bgGreen: "gkcss0-bf54id"
};
`;

exports[`function.js (prod): css 1`] = `.gkcss0-bf54id{background-color:green}`;

exports[`function.js (prod): dependencies 1`] = `Array []`;

exports[`import.js (dev): compiled 1`] = `
import * as Colors from "./imports/colors.es";

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: Colors.GREEN
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

import * as Colors from "./imports/colors.es";
const styles = {
  bgGreen: "gkcss0-backgroundColor-green"
};
`;

exports[`import.js (dev): css 1`] = `.gkcss0-backgroundColor-green{background-color:green}`;

exports[`import.js (dev): dependencies 1`] = `
Array [
  /Users/dual/Code/geekie/css/tests/babel/fixtures/imports/colors.es.js,
]
`;

exports[`import.js (prod): compiled 1`] = `
import * as Colors from "./imports/colors.es";

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: Colors.GREEN
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

import * as Colors from "./imports/colors.es";
const styles = {
  bgGreen: "gkcss0-bf54id"
};
`;

exports[`import.js (prod): css 1`] = `.gkcss0-bf54id{background-color:green}`;

exports[`import.js (prod): dependencies 1`] = `
Array [
  /Users/dual/Code/geekie/css/tests/babel/fixtures/imports/colors.es.js,
]
`;

exports[`import-multiple-specifiers.js (dev): compiled 1`] = `
import { GREEN, RED } from "./imports/colors.es";

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: GREEN
  },
  bgRed: {
    backgroundColor: RED
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { GREEN, RED } from "./imports/colors.es";
const styles = {
  bgGreen: "gkcss0-backgroundColor-green",
  bgRed: "gkcss0-backgroundColor-red"
};
`;

exports[`import-multiple-specifiers.js (dev): css 1`] = `
.gkcss0-backgroundColor-green{background-color:green}
.gkcss0-backgroundColor-red{background-color:red}
`;

exports[`import-multiple-specifiers.js (dev): dependencies 1`] = `
Array [
  /Users/dual/Code/geekie/css/tests/babel/fixtures/imports/colors.es.js,
]
`;

exports[`import-multiple-specifiers.js (prod): compiled 1`] = `
import { GREEN, RED } from "./imports/colors.es";

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: GREEN
  },
  bgRed: {
    backgroundColor: RED
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { GREEN, RED } from "./imports/colors.es";
const styles = {
  bgGreen: "gkcss0-bf54id",
  bgRed: "gkcss0-5scuol"
};
`;

exports[`import-multiple-specifiers.js (prod): css 1`] = `
.gkcss0-bf54id{background-color:green}
.gkcss0-5scuol{background-color:red}
`;

exports[`import-multiple-specifiers.js (prod): dependencies 1`] = `
Array [
  /Users/dual/Code/geekie/css/tests/babel/fixtures/imports/colors.es.js,
]
`;

exports[`require.js (dev): compiled 1`] = `
const colors = require("./imports/colors.cjs");

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: colors.GREEN
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const colors = require("./imports/colors.cjs");

const styles = {
  bgGreen: "gkcss0-backgroundColor-green"
};
`;

exports[`require.js (dev): css 1`] = `.gkcss0-backgroundColor-green{background-color:green}`;

exports[`require.js (dev): dependencies 1`] = `
Array [
  /Users/dual/Code/geekie/css/tests/babel/fixtures/imports/colors.cjs.js,
]
`;

exports[`require.js (prod): compiled 1`] = `
const colors = require("./imports/colors.cjs");

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: colors.GREEN
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const colors = require("./imports/colors.cjs");

const styles = {
  bgGreen: "gkcss0-bf54id"
};
`;

exports[`require.js (prod): css 1`] = `.gkcss0-bf54id{background-color:green}`;

exports[`require.js (prod): dependencies 1`] = `
Array [
  /Users/dual/Code/geekie/css/tests/babel/fixtures/imports/colors.cjs.js,
]
`;

exports[`simple-var.js (dev): compiled 1`] = `
const green = "green";

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: green
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const green = "green";
const styles = {
  bgGreen: "gkcss0-backgroundColor-green"
};
`;

exports[`simple-var.js (dev): css 1`] = `.gkcss0-backgroundColor-green{background-color:green}`;

exports[`simple-var.js (dev): dependencies 1`] = `Array []`;

exports[`simple-var.js (prod): compiled 1`] = `
const green = "green";

const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: green
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const green = "green";
const styles = {
  bgGreen: "gkcss0-bf54id"
};
`;

exports[`simple-var.js (prod): css 1`] = `.gkcss0-bf54id{background-color:green}`;

exports[`simple-var.js (prod): dependencies 1`] = `Array []`;

exports[`static.js (dev): compiled 1`] = `
const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: "green"
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = {
  bgGreen: "gkcss0-backgroundColor-green"
};
`;

exports[`static.js (dev): css 1`] = `.gkcss0-backgroundColor-green{background-color:green}`;

exports[`static.js (dev): dependencies 1`] = `Array []`;

exports[`static.js (prod): compiled 1`] = `
const styles = StyleSheet.create({
  bgGreen: {
    backgroundColor: "green"
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = {
  bgGreen: "gkcss0-bf54id"
};
`;

exports[`static.js (prod): css 1`] = `.gkcss0-bf54id{background-color:green}`;

exports[`static.js (prod): dependencies 1`] = `Array []`;
